docs/styleguide-cpp.md

```diff
@@ -1,320 +1,660 @@
+| 15     | Filenames                         | Lower snake_case                    | my_cool_example.hpp/my_cool_example.cpp |
+
+### 1.4. Do’s and Don’ts
+
+#### 1.4.1. Keep the opening scope brace on the same line
+
+```c++
+//Correct
+if (condition) {
+    //Cool things here
+}
+
+//Incorrect
+if (condition)
+{
+    //cool things here
+}
+```
+
+#### 1.4.2. Keep spaces between arithmetic operators, relational operators, logical operators, variables, and literals
+
+```c++
+//Correct
+int my_number = (some_variable + 10) % 3;
+
+//Incorrect
+int my_number=(some_variable+10)%3;
+```
+
+#### 1.4.3. Prefix and postfix operators should be immediately attached to the variable
+
+```c++
+//Correct
+for (int x = 0; x < 10; x++) {
+    //Something cool
+}
+
+//Incorrect
+for (int x = 0; x < 10; x ++ ) {
+    //Something cool
+}
+```
+
+#### 1.4.4. Avoid `inline`, use Godot’s `_FORCE_INLINE_` for release inline behavior and `_ALWAYS_INLINE_` for always inline behavior
+
+Header
+
+`OpenVic2/godot-cpp/include/godot-cpp/core/defs.hpp`
+
+##### Why?
+
+Godot has inbuilt macros for more compiler consistent inline behavior,
+as well inline can make debugging a lot more difficult and so should
+generally be avoided in debug most of the time.
+
+```c++
+//Correct
+_FORCE_INLINE_ void inline_in_release_build() {}
+_ALWAYS_INLINE_ void inline_in_release_and_debug_build() {}
+
+//Incorrect
+inline void inline_in_release_and_debug_build() {}
+```
+
+#### 1.4.5. Avoid function definitions in header files, prefer definitions in implementation files
+
+```c++
+//In Example.hpp
+class Example {
+public:
+    //Prefer
+    void declared_in_header();
+
+    //Avoid
+    void defined_in_header() {}
+};
+
+//In Example.cpp
+Example::declared_in_header() {}
+```
+
+#### 1.4.6. The content of classes should appear in the following order:
+
+1.  Type Aliases
+
+2.  Static (class-wide) Member Variables
+
+3.  Members Variables
+
+4.  Constructors and Destructors
+
+5.  Static (class-wide) Methods
+
+6.  Methods
+
+```c++
+class Human {
+public:
+    using name_t = std::string;
+
+    name_t name;
+    size_t age;
+    size_t family_members;
+
+    void cool_public_method();
+    void another_cool_public_method();
+
+    _FORCE_INLINE_ bool is_voting_age() const {
+        return age >= 18;
+    }
+
+private:
+    static size_t _number_of_humans;
+
+    Human();
+    ~Human();
+
+    static bool is_more_humans_than(size_t number);
+
+    void cool_private_method();
+};
+```
+
+#### 1.4.7. Class visibility specifiers should be the same indentation of the class
+
+```c++
+class ExampleClass {
+public:
+    int example_public_variable;
+
+protected:
+    int example_protected_variable;
+
+private:
+    int example_private_variable;
+};
+```
+
+#### 1.4.8. Class visibility specifiers should be followed by a declaration on a new line at the next indentation level
```

Comment:
Also totally subjective change

Reply:
It distinguished visibility clearly, else it blends in with the rest of the class.

Reply:
> Also totally subjective change

@FakeByte is correct, this is merely a matter of preference

Reply:
Calling it subjective doesn't resolve anything if there is a reason behind the change.

Reply:
I also prefer having access specifiers an indent level lower than the definitions following them, it helps break up the code into sections rather than one big block.